src/core/thread/network_diagnostic_tlvs.hpp

```diff
@@ -1181,7 +1189,7 @@ class ChildTableEntry
      * @returns The Child ID value.
      *
      */
-    uint16_t GetChildId(void) const { return HostSwap16(mTimeoutRsvChildId) & kChildIdMask; }
+    uint16_t GetChildId(void) const { return ((mFields.m8[0] & kChildIdMask) | mFields.m8[1]); }
```

Comment:
Is it not possible to keep the `HostSwap*` macros while keeping `scan-build` happy?  For me, this makes the code less readable.